From e1b6ac5c6ca3db037127f6be2721fa902a1bbd81 Mon Sep 17 00:00:00 2001
From: Vadim Zhukov <persgray@gmail.com>
Date: Wed, 6 May 2020 17:23:08 +0300
Subject: [PATCH 5/5] QMAKE_POST_LINK doesn't work in Qt 5.5. (#10)

Replace QMAKE_POST_LINK with more extensive usage of INSTALLS.
---
 qmltermwidget.pro | 21 ++++++++++++++-------
 1 file changed, 14 insertions(+), 7 deletions(-)

diff --git a/qmltermwidget.pro b/qmltermwidget.pro
index 128cb30..8b159b0 100644
--- a/qmltermwidget.pro
+++ b/qmltermwidget.pro
@@ -25,12 +25,6 @@ OTHER_FILES += \
     test-app/test-app.qml \
     src/qmldir
 
-# Copy the files useful to the plugin in DESTDIR
-QMAKE_POST_LINK = $(COPY_DIR) $$PWD/lib/color-schemes $$DESTDIR && \
-    $(COPY_DIR) $$PWD/lib/kb-layouts $$DESTDIR && \
-    $$QMAKE_COPY $$PWD/src/qmldir $$DESTDIR && \
-    $$QMAKE_COPY $$PWD/src/QMLTermScrollbar.qml $$DESTDIR
-
 #########################################
 ## INTALLS
 #########################################
@@ -49,4 +43,17 @@ assets.path += $$INSTALL_DIR/$$PLUGIN_IMPORT_PATH
 qmldir.files += $$PWD/src/qmldir
 qmldir.path += $$INSTALL_DIR/$$PLUGIN_IMPORT_PATH
 
-INSTALLS += target qmldir assets
+colorschemes.files = $$PWD/lib/color-schemes/*
+colorschemes.path = $$INSTALL_DIR/$$PLUGIN_IMPORT_PATH/color-schemes
+colorschemes2.files = $$PWD/lib/color-schemes/historic/*
+colorschemes2.path = $$INSTALL_DIR/$$PLUGIN_IMPORT_PATH/color-schemes/historic
+
+kblayouts.files = $$PWD/lib/kb-layouts/*
+kblayouts.path = $$INSTALL_DIR/$$PLUGIN_IMPORT_PATH/kb-layouts
+kblayouts2.files = $$PWD/lib/kb-layouts/historic/*
+kblayouts2.path = $$INSTALL_DIR/$$PLUGIN_IMPORT_PATH/kb-layouts/historic
+
+scrollbar.files = $$PWD/src/QMLTermScrollbar.qml
+scrollbar.path = $$INSTALL_DIR/$$PLUGIN_IMPORT_PATH
+
+INSTALLS += target qmldir assets colorschemes colorschemes2 kblayouts kblayouts2 scrollbar
-- 
2.27.0

